### DISCUSSION TITLE
Why would someone use cmdlets other than Import-CommandConfiguration

### DISCUSSION THREAD

**Discussion by kvprasoon** [PowerShell/Crescendo#73] [Q&A]
If Crescendo is to create a PowerShell module for an executable using a json configuration, Why would someone use below cmdlets ?
```
Export-Schema
Import-CommandConfiguration
New-CrescendoCommand
New-ExampleInfo
New-ParameterInfo
New-UsageInfo
```
Just a doubt.

**Comment by theJasonHelmick** [comment]
;) Great point. So, because we are still early in the process, there are artifacts that we may not continue moving forward. Many of these cmdlets you mention come from an early prototype that @JamesWTruher was working on and may be dropped.

**Reply by kvprasoon** [reply]
I have a VOD session @ PSSummit 21 and is about Crescendo. About to do the recording. So thought of getting clarity :-)

**Reply by theJasonHelmick** [reply]
Jim and I are doing one too - thanks for the help getting the word out!  In the next couple of previews as we add features we will evaluate the cmdlets we need.  Thanks for the patience.

**Reply by ethanbergstrom** [reply]
`New-CrescendoCommand` definitely has its uses for creating and manipulating Command objects natively without having to touch JSON. See my comment [here](https://github.com/PowerShell/Crescendo/discussions/28#discussioncomment-506767).

Now we just need a way to pass that Command object directly to Export-CrescendoModule instead of converting to JSON only to convert it back...

**Reply by kvprasoon** [reply]
You are right.

I was thinking more from a lazy end users perspective :-). Those other commands can be used to develop another f/w to create a DSL for Crescendo.
